### ISSUE TITLE
sigmoid cross entropy doesn't verify shapes

### ISSUE THREAD

**Issue by kshmelkov** [tensorflow/tensorflow#1343]
Hello,

I have found a misleading behaviour in `sigmoid_cross_entropy_with_logits` function. If I pass a tensor of shape `(n, 1)` as logits and a tensor of shape `(n,)` as targets it outputs a tensor of `(n, n)` which is obviously different from tensor `(n,)` if both tensors have the same shape `(n,)`. Although api docs say that logits and targets must have the same shape, this example doesn't fail properly nor broadcast correctly.

OS: Fedora 21
TF v0.7.1 compiled from sources (tagged commit).

**Comment by vrv** [comment]
That is indeed a bug.  We can add the proper shape validation code in that function to prevent the unexpected broadcasting.
